File: HashTable/MyHashMap.py
class MyHashMap(object):

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.hash_map = [None] * 1000
        

    def put(self, key, value):
        """
        value will always be non-negative.
        :type key: int
        :type value: int
        :rtype: None
        """
        the_hash = hash(key) % 1000
        if not self.hash_map[the_hash]:
            self.hash_map[the_hash] = []
        self.remove(key)
        self.hash_map[the_hash].append((key, value))
        

    def get(self, key):
        """
        Returns the value to which the specified key is mapped, or -1 if this map contains no mapping for the key
        :type key: int
        :rtype: int
        """
        the_hash = hash(key) % 1000
        
        if self.hash_map[the_hash]:
            for t in self.hash_map[the_hash]:
                if t[0] == key:
                    return t[1]
        return -1

    def remove(self, key):
        """
        Removes the mapping of the specified value key if this map contains a mapping for the key
        :type key: int
        :rtype: None
        """
        the_hash = hash(key) % 1000
        to_delete = None
        if self.hash_map[the_hash]:
            new_cell = []
            for i in range(len(self.hash_map[the_hash])):
                if not self.hash_map[the_hash][i][0] == key:
                    new_cell.append(self.hash_map[the_hash][i])                    
            self.hash_map[the_hash] = new_cell

File: HashTable/MyHashMap.java
class MyHashMap {
    
    ArrayList<ArrayList<ArrayList<Integer>>> hashMap = new ArrayList<ArrayList<ArrayList<Integer>>>(1000);
    
    /** Initialize your data structure here. */
    public MyHashMap(){
        for(int i = 0; i<1000; i++){
            ArrayList<ArrayList<Integer>> to_add = new ArrayList<ArrayList<Integer>>();
            to_add.add(new ArrayList<Integer>());
            hashMap.add(to_add);
        }
    }
    
    /** value will always be non-negative. */
    public void put(int key, int value) {
        int theHash = key % 1000;
        if (hashMap.get(theHash) == null){
            hashMap.set(theHash, new ArrayList<ArrayList<Integer>>());
        }
        remove(key);       
        hashMap.get(theHash).add(new ArrayList<Integer>());
        hashMap.get(theHash).get(hashMap.get(theHash).size()-1).add(0,key);
        hashMap.get(theHash).get(hashMap.get(theHash).size()-1).add(1,value); 
    }
    
    /** Returns the value to which the specified key is mapped, or -1 if this map contains no mapping for the key */
    public int get(int key) {
        int theHash = key % 1000;
        if (hashMap.get(theHash) != null){
            System.out.println(hashMap.get(theHash));
            for (int i = 0 ; i < hashMap.get(theHash).size(); i++){
                    if (hashMap.get(theHash).get(i).size() == 2){
                        if (hashMap.get(theHash).get(i).get(0) == key){
                            return hashMap.get(theHash).get(i).get(1);
                        }
                    }
            }
        }
        return -1;
    }
    
    /** Removes the mapping of the specified value key if this map contains a mapping for the key */
    public void remove(int key) {
        int theHash = key % 1000;
        if (hashMap.get(theHash) != null){
            ArrayList<ArrayList<Integer>> newCell = new ArrayList<ArrayList<Integer>>();
            newCell.add(new  ArrayList<Integer>());
            for (int i = 0 ; i < hashMap.get(theHash).size(); i++){
                    if (hashMap.get(theHash).get(i).size() == 2){
                        if (hashMap.get(theHash).get(i).get(0) != key){
                            ArrayList<Integer> aCell = new ArrayList<Integer>();
                            aCell.add(hashMap.get(theHash).get(i).get(0));
                            aCell.add(hashMap.get(theHash).get(i).get(1));    
                            newCell.add(aCell);
                        }
                    }
            }
            hashMap.set(theHash, newCell);
        }
        
    }
}
